arrow_expts/spine/dbmap.py

```diff
@@ -25,31 +26,64 @@ def filter_frequencies(freq: str) -> str:
         .replace("year", "Y") \
         .replace("months", "M") \
         .replace("month", "M") \
-        .replace("quarters", "Q") \
-        .replace("quarter", "Q") \
         .replace("weeks", "W") \
         .replace("week", "W") \
+        .replace("days", "D") \
+        .replace("day", "D") \
         .replace("hours", "h") \
         .replace("hour", "h") \
         .replace("minutes", "min") \
         .replace("minute", "min") \
         .replace("seconds", "s") \
-        .replace("second", "s") \
-        .replace("microseconds", "us") \
-        .replace("microsecond", "us") \
-        .replace("nanoseconds", "ns") \
-        .replace("nanosecond", "ns")
+        .replace("second", "s")
     if re.compile("^[0-9]+$").match(filtered_freq):
         # If frequency is an integer, the implied unit is "minutes"
-        return freq + "m"
+        return filtered_freq + "m"
     else:
-        return freq
+        return filtered_freq
 
 class IndexType(Enum):
     Timestamp = auto()
     Sequence = auto()
     Generic = auto()
 
+to_numpy = {
+    "Y": "Y",
+    "M": "M",
+    "W": "W",
+    "D": "D",
+    "h": "h",
+    "min": "m",
+    "s": "s",
+}
+
+def low_res_datetime(start: str, freq: str, periods: int) -> pd.DatetimeIndex:
+    """Create pd.DatetimeIndex with lower time resolution.
+
+    The default resolution of pd.date_time is [ns], which puts boundaries on allowed start- and end-dates due to limited storage capacity. Choosing a resolution of [s] instead opens up that range considerably.
+
+    "For nanosecond resolution, the time span that can be represented using a 64-bit integer is limited to approximately 584 years."
+    - https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#timestamp-limitations
+
+    You can check the available ranges with `pd.Timestamp.min` and `pd.Timestamp.max`.
+    """
+    period_parts = re.search(r'^([0-9]+) *(.*)$', freq).groups()
+    if len(period_parts) != 2:
+        raise ValueError("Can't analyze period \"{period}\".")
```

Comment:
Error message is not specific.  Say what is the problem, like couldn't separate number from unit, etc.